### PR TITLE
Add the filestats receiver

### PR DESCRIPTION
**Pull request by pjanotti**
Adding the `filestats` receiver from contrib.

### CODE REVIEW

File: tests/receivers/filestats/testdata/all_expected_metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+              dataPoints:
+                - asInt: "3"
+                  startTimeUnixNano: "1000000"
+                  timeUnixNano: "2000000"
+            name: file.size
+            unit: b
+        scope:
+          name: github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filestatsreceiver
+          version: latest
+  - resource:
```
Comment (dmitryax) [inline tests/receivers/filestats/testdata/all_expected_metrics.yaml:32]: Looks like an extra redundant resource after this

**Review by dmitryax — APPROVED** [review]

File: tests/receivers/filestats/filestats_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	"testing"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestFileStatsIntegration(t *testing.T) {
```
Comment (dmitryax) [inline tests/receivers/filestats/filestats_test.go:27]: There is a bit more sophisticated test upstream https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/filestatsreceiver/integration_test.go. Why do we need to bring it here?

Comment (pjanotti) [reply tests/receivers/filestats/filestats_test.go:27]: Doh! Removing it

Comment (pjanotti) [reply tests/receivers/filestats/filestats_test.go:27]: I think there may be another one or two that we can drop, I will look at those.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**